dftimewolf/lib/collectors/gcloud.py

```diff
@@ -137,43 +154,48 @@ def SetUp(self,
           image_project=image_project,
           image_family=image_family)
 
-      if disk_names:
-        for name in disk_names:
+    except AccessTokenRefreshError as err:
+      self.state.AddError('Something is wrong with your gcloud access token.')
+      self.state.AddError(err, critical=True)
+
+    except ApplicationDefaultCredentialsError as err:
+      self.state.AddError(
+          'Something is wrong with your Application Default Credentials. '
+          'Try running:\n  $ gcloud auth application-default login')
+      self.state.AddError(err, critical=True)
+
+  def FindDisksToCopy(self):
+    """Determines which disks to copy depending on object attributes."""
+    if not (self.remote_instance_name or self.disk_names):
+      self.state.AddError(
+          'You need to specify at least an instance name or disks to copy',
+          critical=True)
+      return
+
+    try:
```

Comment:
This is pretty heavily nested and hard to follow, can you please refactor this? Maybe it's worth breaking out some functions here - _GetDisksFromNames() maybe?